packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -188,6 +189,9 @@ function nodeDataToMapTree(
 	// They already have the mapTree, so there is no need to recompute it.
 	const flexNode = tryGetFlexNode(data);
 	if (flexNode !== undefined) {
+		if (!allowedTypes.has(tryGetSimpleNodeSchema(flexNode.schema) ?? fail("missing schema"))) {
```

Comment:
We should probably put this error inside of the `isMapTreeNode` check. That means that if a user tries to insert a tree that is already inserted _and_ is out of schema, they will get the "already inserted" error rather than the "out of schema" error. I think the former is more fundamental because it indicates the user is doing something categorically wrong (re-inserting trees is always illegal, everywhere) rather than merely wrong in this context (the schema doesn't match up here, but it probably does somewhere else).

Reply:
Moved usage error to inside of the `isMapTreeNode` check